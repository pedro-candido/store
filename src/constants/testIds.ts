const testIds = {
  addToCartButton: 'add-to-cart-button',
  removeFromCartButton: 'remove-from-cart-button',
  productTestId: 'product-',
  backButton: 'back-button',
  cartButton: 'cart-button',
};

export { testIds };
