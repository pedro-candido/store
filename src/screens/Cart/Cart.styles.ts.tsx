import styled from 'styled-components/native';

import Label from '@components/Label';

export const Container = styled.View`
  height: 95%;
  width: 100%;
  justify-content: space-around;
`;

export const Title = styled(Label)`
  font-size: 16px;
  font-family: 'Gabarito-Medium';
`;
