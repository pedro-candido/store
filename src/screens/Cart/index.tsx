import AddedProductsList from '@components/AddedProductsList';
import { Container, Title } from '@screens/Cart/Cart.styles.ts';
import { useStore } from '@store/globalStore';

function Cart() {
  const { productsChosen } = useStore();

  return (
    <Container>
      {productsChosen.length === 0 && <Title>Ops, não há produtos</Title>}
      {productsChosen.length !== 0 && <AddedProductsList />}
    </Container>
  );
}

export default Cart;
