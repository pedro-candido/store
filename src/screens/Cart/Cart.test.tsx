import * as React from 'react';

import { fireEvent, render, screen } from '@testing-library/react-native';

import { testIds } from '@/constants/testIds';

import Cart from './';

const mockedNavigate = jest.fn;

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => ({
      navigate: mockedNavigate,
      dispatch: jest.fn(),
    }),
    useRoute: () => ({
      params: {
        productId: 1,
      },
    }),
  };
});

describe('Cart', () => {
  it('should copy screen', () => {
    const testrenderer = render(<Cart />);

    expect(testrenderer.toJSON()).toMatchSnapshot();
  });
});
