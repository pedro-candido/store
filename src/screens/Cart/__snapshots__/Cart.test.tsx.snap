// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should copy screen 1`] = `
<View
  style={
    {
      "height": "95%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Gabarito-regular",
        },
        {
          "fontFamily": "Gabarito-Medium",
          "fontSize": 16,
        },
      ]
    }
  >
    Ops, não há produtos
  </Text>
</View>
`;
